src/ogc/wcs/WcsCoverage.js

```diff
@@ -72,13 +74,62 @@ define([
              * A configuration object for use by TiledElevationCoverage.
              * @type {{}}
              */
-            this.elevationConfiguration = this.createElevationConfiguration();
+            this.elevationConfig = this.createElevationConfig();
+        };
+
+        /**
+         * Preferred formats used for fuzzy comparison to available formats.
+         * @type {string[]}
+         */
+        WcsCoverage.PREFERRED_FORMATS = ["geotiff", "tiff"];
+
+        /**
+         * The default data format.
+         * @type {string}
+         */
+        WcsCoverage.DEFAULT_FORMAT = "image/tiff";
+
+        // Internal use only
+        WcsCoverage.prototype.createElevationConfig = function () {
+            return {
+                resolution: this.resolution,
+                coverageSector: this.sector,
+                retrievalImageFormat: this.determineFormatFromService(),
+                urlBuilder: new WcsUrlBuilder(this)
+            };
         };
 
         // Internal use only
-        WcsCoverage.prototype.createElevationConfiguration = function () {
-            // TODO
-            return {};
+        WcsCoverage.prototype.determineFormatFromService = function () {
+            var version = this.service.capabilities.version, availableFormats, format, coverageDescription;
+
+            // find the service supported format identifiers
+            if (version === "1.0.0") {
+                // find the associated coverage description
+                for (var i = 0, len = this.service.coverageDescriptions.coverages.length; i < len; i++) {
+                    if (this.coverageId === this.service.coverageDescriptions.coverages[i].name) {
+                        availableFormats = this.service.coverageDescriptions.coverages[i].supportedFormats.formats;
+                        break;
+                    }
+                }
+            } else if (version === "2.0.1" || version === "2.0.0") {
+                availableFormats = this.service.capabilities.serviceMetadata.formatsSupported
```

Comment:
This line is missing the terminating semicolon. Please double-check this hasn't occurred anywhere else.

Reply:
Seems I missed a number of semicolons, thanks for the check